glide-core/telemetry/src/metrics_exporter_file.rs

```
@@ -0,0 +1,226 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use opentelemetry_sdk::metrics::data::ResourceMetrics;
+use opentelemetry_sdk::metrics::data::{Gauge, Histogram, Sum};
+use opentelemetry_sdk::metrics::exporter::PushMetricExporter;
+use opentelemetry_sdk::metrics::MetricError;
+use opentelemetry_sdk::metrics::MetricResult;
+use opentelemetry_sdk::metrics::Temporality;
+use serde_json::{Map, Value};
+use std::any::Any;
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::path::PathBuf;
+use std::sync::atomic;
+
+/// An OpenTelemetry exporter that writes Metrics to a file on export.
+pub struct FileMetricExporter {
+    is_shutdown: atomic::AtomicBool,
+    path: PathBuf,
+}
+
+impl FileMetricExporter {
+    pub fn new(path: PathBuf) -> std::result::Result<Self, MetricError> {
+        Ok(FileMetricExporter {
+            is_shutdown: atomic::AtomicBool::new(false),
+            path,
+        })
+    }
+}
+
+#[async_trait]
+impl PushMetricExporter for FileMetricExporter {
+    fn temporality(&self) -> Temporality {
+        Temporality::Cumulative
+    }
+
+    async fn export(&self, metrics: &mut ResourceMetrics) -> MetricResult<()> {
+        println!("Start to export:: metrics: {:?}", metrics);
+        if self.is_shutdown.load(atomic::Ordering::SeqCst) {
+            return Ok(());
```

shouldn't we return an error to the caller that tries to export on a shutdown exporter?

---

added. - return an error from ```export``` if called after shutdown